x/auth/client/cli/cli_test.go

```diff
@@ -1024,6 +1024,85 @@ func (s *IntegrationTestSuite) createBankMsg(val *network.Validator, toAddr sdk.
 	return res
 }
 
+func (s *IntegrationTestSuite) TestSignWithMultiSigners() {
+	val0, val1 := s.network.Validators[0], s.network.Validators[1]
+	val0Coin := sdk.NewCoin(fmt.Sprintf("%stoken", val0.Moniker), sdk.NewInt(10))
+	val1Coin := sdk.NewCoin(fmt.Sprintf("%stoken", val1.Moniker), sdk.NewInt(10))
+	_, _, addr1 := testdata.KeyTestPubAddr()
+
+	// Creating a tx with 2 msgs from 2 signers: val0 and val1.
+	txBuilder := val0.ClientCtx.TxConfig.NewTxBuilder()
+	txBuilder.SetMsgs(
+		banktypes.NewMsgSend(val0.Address, addr1, sdk.NewCoins(val0Coin)),
+		banktypes.NewMsgSend(val1.Address, addr1, sdk.NewCoins(val1Coin)),
+	)
+	txBuilder.SetFeeAmount(sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))))
+	txBuilder.SetGasLimit(testdata.NewTestGasLimit())
+	// Set signer_infos for both signers. Note: we use the empty signature hack.
+	txBuilder.SetSignatures(
+		signing.SignatureV2{
+			PubKey:   val0.PubKey,
```

Comment:
I'm pretty sure this is wrong.
In the `val0.ClientCtx` (this is what's used in the CLI, when getting the user key), the keys are different then the `val0.PubKey`:
* `val0.PubKey` - is the ed25519 for block validation
* `valInfo, err := val1.ClientCtx.Keyring.Key(val1.Moniker); valInfo.PubKey` is used in a keyring for signing messages.

I've spent a bit of time when writing tests. If you will validate signatures after the `CLI sign` it should fail with the setting you use here. Maybe that's the reason broadcast fails.

Reply:
You can also print here `val0,PubKey` to inspect it more.